proposals/collection-expression-better-conversion.md

```diff
@@ -0,0 +1,125 @@
+- `T'₁`:
+    - If `T₁` is a generic type, `T₁` with all type parameters unsubstituted
+    - If `T₁` is an array type, an unsubstituted array type
+    - Otherwise, `T₁`
+- `T'₂` is the same as `T'₁`, except for using `T₂` instead of `T₁`
+- `E₁` is the element type of `T₁`
+- `E₂` is the element type of `T₂`
+- `CE₁ᵢ` are the series of conversions from `ELᵢ` to `E₁`
+- `CE₂ᵢ` are the series of conversions from `ELᵢ` to `E₂`
+
+`T'₁` is at least as good as `T'₂` if there is an identity conversion between them.  
+`T'₁` is better than `T'₂` if:
+- `T'₁` is ReadOnlySpan<T>, and `T'₂` is not ReadOnlySpan<T>
+- `T'₂` is Span<T>, and `T'₂` is an *array_or_array_interface*
+- `T'₂` is implicitly convertible to `T'₁`, and `T'₁` is not implicitly convertible to `T'₂`.
+  
+Otherwise, `T'₁` neither as good as nor better than `T₂`.
+
+If `E₁` and `E₂` have an identity conversion, then the element conversions are as good as each other. Otherwise, the element conversions to `E₁` are better than the element conversions to `E₂` if:
+- For every `ELᵢ`, `CE₁ᵢ` is at least as good as `CE₂ᵢ`, and
+- There is at least one i where `CE₁ᵢ` is better than `CE₂ᵢ`
+  
+Otherwise, neither set of element conversions is better than the other, and they are also not as good as each other.  
+Conversion comparisons are made using better conversion from expression if `ELᵢ` is not a spread expression. If `ELᵢ` is a spread expression, we use better conversion from the element type of the spread collection to `E₁` or `E₂`, respectively.
+
+`T₁` is a ***better collection conversion from expression*** than `T₂` if:
+- `T'₁` is better than `T'₂`, and the element conversions to `E₁` are at least as good as the element conversions to `E₂`, or
+- `T'₁` is at least as good as `T'₂`, and the element conversions to `E₁` are better than the element conversions to `E₂`.
+
+### Scenarios:
+
+In plain English, the collection types themselves must be either the same, or unambiguously better (ie, `List<T>` and `List<T>` are the same, `List<T>` is unambiguously better than `IEnumerable<T>`, and `List<T>` and `HashSet<T>` cannot be compared), and
+the element conversions for the better collection type must also be the same or better (ie, we can't decide between `ReadOnlySpan<object>` and `Span<string>` for `[""]`, the user needs to make that decision). More examples of this are:
+
+
+| `T₁` | `T₂` | `E` | `C₁` Conversions | `C₂` Conversions | `T'₁` vs `T'₂` | `CE₁ᵢ` vs `CE₂ᵢ` | Outcome |
+|--------|--------|------------|----------------|----------------|---------------------|---------------------|---------|
+| `List<int>` | `List<byte>` | `[1, 2, 3]` | `[Identity, Identity, Identity]` | `[Implicit Constant, Implicit Constant, Implicit Constant]` | `T'₁` is as good as `T'₂` | `CE₁ᵢ` is better | `List<int>` is picked |
+| `List<int>` | `List<byte>` | `[(int)1, (byte)2]` | `[Identity, Implicit Numeric]` | Not applicable | `T'₁` is as good as `T'₂` | `T₂` is not applicable | `List<int>` is picked |
```

Comment:
It looks like `-` is used instead of `Not applicable` in the other rows. For clarity, what do you think of using `N/A` everywhere?